src/bin/spin.rs

```diff
@@ -42,15 +41,14 @@ async fn main() {
 }
 
 async fn _main() -> anyhow::Result<()> {
-    tracing_subscriber::fmt()
-        .with_writer(std::io::stderr)
-        .with_env_filter(
-            tracing_subscriber::EnvFilter::from_default_env()
-                .add_directive("wasmtime_wasi_http=warn".parse()?)
-                .add_directive("watchexec=off".parse()?),
-        )
-        .with_ansi(std::io::stderr().is_terminal())
-        .init();
+    let telemetry_config = match spin_telemetry::config::Config::from_env(VERSION.to_string()) {
+        Ok(config) => config,
+        Err(err) => {
+            eprintln!("Failed to load telemetry config: {err}");
+            return Ok(());
```

Comment:
This exits with success if it fails to load the telemetry config: this means that scripts that rely on the exit code will assume that Spin ran okay.  It should either continue without telemetry, or return an error (which Rust will turn into exit code 1).

Reply:
Also, to print an error explicitly, prefer `terminal::error!` unless there is a compelling reason not to - this provides consistent formatting across the Spin UI.

Reply:
This wasn't intentional to quell the error. I'll make the updates you suggested.